Rubberduck.VBEEditor/SafeComWrappers/Office.Core/CommandBar.cs

```diff
@@ -164,5 +165,31 @@ public override int GetHashCode()
         {
             return IsWrappingNullReference ? 0 : HashCode.Compute(Type, Id, Index, IsBuiltIn, Target.Parent);
         }
+
+        private bool _isReleased;
+        public virtual void Release(bool final = false)
+        {
+            if (IsWrappingNullReference || _isReleased || !Marshal.IsComObject(Target))
+            {
+                _isReleased = true;
+                return;
+            }
+
+            try
+            {
+                if (final)
+                {
+                    Marshal.FinalReleaseComObject(Target);
+                }
+                else
+                {
+                    Marshal.ReleaseComObject(Target);
+                }
+            }
+            finally
+            {
+                _isReleased = true;
```

Comment:
I wonder if we really should be setting it to `true` irrespective of the returns from the `Marshal.FinalReleaseComObject` or `Marshal.ReleaseComObject`? If it's meant to ensure that we should not use the object from the .NET side, that might be a case for implementing the `IDisposable` pattern? Otherwise, we're kind of lying because a non-zero return from either `Marshal.FinalReleaseComObject` or `Marshal.ReleaseComObject` doesn't mean the COM object is in fact released.

Reply:
@bclothier this may or may not be why reloading RD after unloading bricks the VBE... thoughts?